pkg/kubelet/stats/stats_provider_linux.go

```diff
@@ -0,0 +1,48 @@
+// +build linux
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package stats
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	statsapi "k8s.io/kubernetes/pkg/kubelet/apis/stats/v1alpha1"
+)
+
+func (p *StatsProvider) RlimitStats() (*statsapi.RlimitStats, error) {
+	rlimit := &statsapi.RlimitStats{}
+
+	if content, err := ioutil.ReadFile("/proc/sys/kernel/pid_max"); err == nil {
+		if maxPid, err := strconv.ParseInt(string(content[:len(content)-1]), 10, 64); err == nil {
+			rlimit.MaxPID = &maxPid
+		}
+	}
+
+	if files, err := filepath.Glob("/proc/[0-9]*"); err == nil {
```

Comment:
looks like syscall.Sysinfo also reports the number of processes. want to use that?

```
package main

import (
    "fmt"
    "syscall"
)

func main() {
    var info syscall.Sysinfo_t
    syscall.Sysinfo(&info);
    fmt.Println("procs : ", info.Procs)
}

```

Reply:
That's better, let me try it :).

Reply:
it seems there's some gap between two way:

```
root@ubuntu-xenial:/home/ubuntu# ./main
procs :  140
procs :  120
```

```go
package main

import (
	"fmt"
	"path/filepath"
	"syscall"
)

func main() {
	var info syscall.Sysinfo_t
	syscall.Sysinfo(&info)
	fmt.Println("procs : ", info.Procs)

	if files, err := filepath.Glob("/proc/[0-9]*"); err == nil {
		fmt.Println("procs : ", len(files))
	}
}
```

Reply:
Those are the LWP's i believe they count towards the pid_max too
- http://bencane.com/2011/06/25/checking-the-number-of-lwp-threads-in-linux/ 
- http://opensourceforu.com/2011/08/light-weight-processes-dissecting-linux-threads/
- https://serverfault.com/questions/648287/reduce-pid-max-safely

Reply:
let me ping Go community to confirm :).